src/client/QXmppMixManager.cpp

```diff
@@ -0,0 +1,1382 @@
+///
+/// \var QXmppMixManager::Participation::participantId
+///
+/// ID of the user within the channel.
+///
+/// \var QXmppMixManager::Participation::nickname
+///
+/// Nickname of the user within the channel.
+///
+/// If the server modified the desired nickname, this is the modified one.
+///
+/// \var QXmppMixManager::Participation::subscriptions
+///
+/// Nodes belonging to the joined channel that are subscribed to.
+///
+/// If not all desired nodes could be subscribed, this contains only the subscribed nodes.
+///
+
+///
+/// \typedef QXmppMixManager::Jid
+///
+/// JID of a user or domain.
+///
+
+///
+/// \typedef QXmppMixManager::ChannelJid
+///
+/// JID of a MIX channel.
+///
+
+///
+/// \typedef QXmppMixManager::Nickname
+///
+/// Nickname of the user within a MIX channel.
+///
+/// If the server modified the desired nickname, this is the modified one.
+///
+
+///
+/// \typedef QXmppMixManager::CreationResult
+///
+/// Contains the JID of the created MIX channel or a QXmppError on failure.
+///
+
+///
+/// \typedef QXmppMixManager::ChannelJidResult
+///
+/// Contains the JIDs of all discoverable MIX channels of a MIX service or a QXmppError if it
+/// failed.
+///
+
+///
+/// \typedef QXmppMixManager::ChannelNodeResult
+///
+/// Contains all nodes of the requested MIX channel that can be subscribed by the user or a
+/// QXmppError on failure.
+///
+
+///
+/// \typedef QXmppMixManager::ConfigurationResult
+///
+/// Contains the configuration of the MIX channel or a QXmppError on failure.
+///
+
+///
+/// \typedef QXmppMixManager::InformationResult
+///
+/// Contains the information of the MIX channel or a QXmppError on failure.
+///
+
+///
+/// \typedef QXmppMixManager::JoiningResult
+///
+/// Contains the result of the joined MIX channel or a QXmppError on failure.
+///
+
+///
+/// \typedef QXmppMixManager::NicknameResult
+///
+/// Contains the new nickname within a joined MIX channel or a QXmppError on failure.
+///
+
+///
+/// \typedef QXmppMixManager::InvitationResult
+///
+/// Contains the requested invitation to a MIX channel or a QXmppError on failure.
+///
+
+///
+/// \typedef QXmppMixManager::SubscriptionResult
+///
+/// Contains the result of the subscribed/unsubscribed nodes belonging to a MIX channel or a
+/// QXmppError on failure.
+///
+
+///
+/// \typedef QXmppMixManager::JidResult
+///
+/// Contains the JIDs of users or domains that are allowed to participate resp. banned from
+/// participating in a MIX channel or a QXmppError on failure.
+///
+
+///
+/// \typedef QXmppMixManager::ParticipantResult
+///
+/// Contains the participants of a MIX channel or a QXmppError on failure.
+///
+
+constexpr QStringView MIX_SERVICE_DISCOVERY_NODE = u"mix";
+
+///
+/// Constructs a MIX manager.
+///
+QXmppMixManager::QXmppMixManager()
+    : d(new QXmppMixManagerPrivate())
+{
+}
+
+QXmppMixManager::~QXmppMixManager() = default;
+
+QStringList QXmppMixManager::discoveryFeatures() const
+{
+    return { ns_mix.toString() };
```

Comment:
probably also mix_pam (and maybe other namespaces), right?

Reply:
In fact, the other namespaces are only announced by servers.